server/pulp/server/webservices/views/util.py

```diff
@@ -61,7 +82,8 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
 
         try:
-            request.body_as_json = json.loads(request.body)
+            request_json = json.loads(request.body)
+            request.body_as_json = _ensure_input_encoding(request_json)
```

Comment:
Can you move this line to an `else`? We are trying to have only one line in a `try` so it is clear what caused the exception.